tests/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/Rename/RenamerTestsBase.cs

```diff
@@ -92,36 +93,40 @@ protected override async Task<bool> IsAutomationFunctionAsync()
-            var extensibility = new Mock<IVsUIService<IVsExtensibility, IVsExtensibility3>>().Object;
-            var operationProgressMock = new Mock<IVsService<SVsOperationProgress, IVsOperationProgressStatusService>>().Object;
-            var context = new Mock<IProjectTreeActionHandlerContext>().Object;
+            var refactorNotifyService = Mock.Of<IRefactorNotifyService>();
+            var extensibility = Mock.Of<IVsUIService<IVsExtensibility, IVsExtensibility3>>();
+            var operationProgressMock = Mock.Of<IVsService<SVsOperationProgress, IVsOperationProgressStatusService>>();
+            var context = Mock.Of<IProjectTreeActionHandlerContext>();
 
             var mockNode = new Mock<IProjectTree>();
             mockNode.SetupGet(x => x.FilePath).Returns(oldFilePath);
             mockNode.SetupGet(x => x.IsFolder).Returns(false);
             var node = mockNode.Object;
 
-            var renamer = new TestRenamerProjectTreeActionHandler(unconfiguredProject,
-                                                              projectServices,
-                                                              new Lazy<Workspace>(() => ws),
-                                                              environmentOptionsFactory,
-                                                              userNotificationServices,
-                                                              roslynServices,
-                                                              waitIndicator,
-                                                              vsOnlineServices,
-                                                              projectAsynchronousTasksService,
-                                                              projectThreadingService,
-                                                              extensibility,
-                                                              operationProgressMock,
-                                                              settingsManagerService);
+            // Construct the renamer.
+            var renamer = new TestRenamerProjectTreeActionHandler(
+                unconfiguredProject,
+                projectServices,
+                new Lazy<Workspace>(() => ws),
+                environmentOptionsFactory,
+                userNotificationServices,
+                roslynServices,
+                waitIndicator,
+                vsOnlineServices,
+                projectAsynchronousTasksService,
+                projectThreadingService,
+                extensibility,
+                operationProgressMock,
+                settingsManagerService);
 
-            await renamer.RenameAsync(context, node, newFilePath)
-                         .TimeoutAfter(TimeSpan.FromSeconds(1));
+            // Perform the rename, with a timeout in case of hangs.
+            await renamer
+                .RenameAsync(context, node, newFilePath)
+                .TimeoutAfter(TimeSpan.FromSeconds(30));
```

Comment:
Why have we increased to 30sec? Does rename take longer now? We would still want to timeout sooner correct (> 1 sec but <10sec)

Reply:
The reason for the change is this code now actually yields the thread, which vastly increases the variance on test execution time on a heavily laden CI VM, with tests running concurrently. Previously it was straight-line synchronous code behind the task. Now it yields.

The reason for 30 over 10 is that this timeout is just for unit tests where there's a lot of variance in CI. A timeout of this magnitude can only really be a hang, for which any timeout duration is fine.